src/services/LeafValueCoder.ts

```diff
@@ -12,10 +15,23 @@ export class LeafValueCoder {
    * @param bits
    */
   static encode = (n: FeedValue, label: string, bits = 256): Buffer => {
-    const hex = LeafValueCoder.isFixedValue(label)
-      ? LeafValueCoder.fixedValueToHex(n)
-      : new BigNumber(n, 10).times(NUMERIC_MULTIPLIER).toString(16);
-    return LeafValueCoder.encodeHex(hex, bits);
+    if (LeafValueCoder.isFixedValue(label)) {
+      return LeafValueCoder.encodeHex(LeafValueCoder.fixedValueToHex(n), bits);
+    }
+
+    if (LeafValueCoder.isIntValue(label)) {
+      if (typeof n === 'number' || typeof n === 'bigint') {
+        return LeafValueCoder.encodeHex(LeafValueCoder.intValueToHex(n), bits);
+      } else {
+        throw Error(`${n} is not valid number or bigint`);
+      }
+    }
+
+    if (typeof n === 'number') {
```

Comment:
let's not do that, it should be ok if value is passed as string, also we do not have any issues with that part of the code, let's not change behaviour

Reply:
oh you changed the `FeedValue` type, below line should work for not numbers, so we should not need this condition

Reply:
the BigNumber constructor wont accept bigint as value, but i use the following and all the tests pass:
`return LeafValueCoder.encodeHex(new BigNumber(n.toString(), 10).times(NUMERIC_MULTIPLIER).toString(16), bits);`
(removed the condition on line 30)